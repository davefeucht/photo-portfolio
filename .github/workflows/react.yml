# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
