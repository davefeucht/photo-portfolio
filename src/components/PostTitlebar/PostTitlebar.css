.post-titlebar {
  background-color: #000000;
  font-size: 25px;
  color: #FFFFFF;
  grid-area: header;
  display: grid;
  grid-template-columns: 95% 5%;
  grid-template-rows: 100%;
}

.post-titlebar > .title {
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  margin: auto auto auto 5px;
}