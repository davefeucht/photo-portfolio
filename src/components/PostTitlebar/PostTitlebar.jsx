/** **************
* PostTitlebar component displays the titlebar for a Post
*************** */

import './PostTitlebar.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';

import CloseButton from '../CloseButton/CloseButton.jsx';

function PostTitlebar({ postTitle }) {
    return (
        <div className="post-titlebar">
            <div className="title">{postTitle}</div>
            <CloseButton />
        </div>
    );
}

PostTitlebar.displayName = 'PostTitlebar';

PostTitlebar.propTypes = {
    postTitle: PropTypes.string.isRequired
};

export default observer(PostTitlebar);
