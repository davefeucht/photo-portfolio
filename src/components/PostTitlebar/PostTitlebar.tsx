/** **************
* PostTitlebar component displays the titlebar for a Post
*************** */

import './PostTitlebar.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import CloseButton from '../CloseButton/CloseButton';

interface PostTitlebarProps {
    postTitle: string
}

const PostTitlebar: React.FC<PostTitlebarProps> = ({ postTitle = null }) => {
    return (
        <div className="post-titlebar">
            <div className="title">{postTitle}</div>
            <CloseButton />
        </div>
    );
};

PostTitlebar.displayName = 'PostTitlebar';

export default observer(PostTitlebar);
