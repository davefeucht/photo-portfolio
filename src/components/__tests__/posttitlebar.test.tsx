import { render } from '@testing-library/react';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';

import PostTitlebar from '../PostTitlebar/PostTitlebar';

const postTitle = 'Some Post';

test('PostTitlebar displays', async () => {
    const { container } = render(
        <MemoryRouter>
            <PostTitlebar postTitle={postTitle} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
