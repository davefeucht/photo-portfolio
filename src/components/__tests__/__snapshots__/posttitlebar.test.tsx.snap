// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTitlebar displays 1`] = `
<div
  class="post-titlebar"
>
  <div
    class="title"
  >
    Some Post
  </div>
  <div
    class="close-button"
    role="button"
  >
    <a
      href="/category/undefined"
    >
      <div
        class="close-button-content"
      >
        <img
          height="30px"
          src="./assets/images/close-button.svg"
          width="30px"
        />
      </div>
    </a>
  </div>
</div>
`;
